velox/common/memory/Memory.h

```diff
@@ -85,30 +85,31 @@ class IMemoryManager {
   /// Returns the memory allocation alignment of this memory manager.
   virtual uint16_t alignment() const = 0;
 
-  /// Creates a memory pool for use with specified 'name', 'kind' and 'cap'. If
-  /// 'name' is missing, the memory manager generates a default name internally
-  /// to ensure uniqueness. If 'kind' is kAggregate, a root memory pool is
-  /// created. Otherwise, a leaf memory pool is created as the child of the
-  /// memory manager's default root memory pool.
-  virtual std::shared_ptr<MemoryPool> getPool(
+  /// Creates a root memory pool with specified 'name' and 'cap'. If 'name' is
+  /// missing, the memory manager generates a default name internally to ensure
+  /// uniqueness.
+  virtual std::shared_ptr<MemoryPool> getRootPool(
       const std::string& name = "",
-      MemoryPool::Kind kind = MemoryPool::Kind::kAggregate,
       int64_t maxBytes = kMaxMemory) = 0;
 
+  /// Creates a leaf memory pool for direct memory allocation use with specified
+  /// 'name'. If 'name' is missing, the memory manager generates a default name
```

Comment:
Perhaps, clarify that even if 'name' is provided, it is augmented with some unique value to ensure uniqueness.